src/cairoUtilFuncGen/storage/dynArrayIndexAccess.ts

```diff
@@ -71,29 +71,22 @@ export class DynArrayIndexAccessGen extends StringIndexedFuncGen {
     const funcName = `${arrayName}_IDX`;
     return {
       name: funcName,
-      code: [
-        `func ${funcName}{syscall_ptr : felt*, pedersen_ptr : HashBuiltin*, range_check_ptr : felt}(ref: felt, index: Uint256) -> (res: felt){`,
-        `    alloc_locals;`,
-        `    let (length) = ${lengthName}.read(ref);`,
-        `    let (inRange) = uint256_lt(index, length);`,
-        `    assert inRange = 1;`,
-        `    let (existing) = ${arrayName}.read(ref, index);`,
-        `    if (existing == 0){`,
-        `        let (used) = WARP_USED_STORAGE.read();`,
-        `        WARP_USED_STORAGE.write(used + ${valueCairoType.width});`,
-        `        ${arrayName}.write(ref, index, used);`,
-        `        return (used,);`,
-        `    }else{`,
-        `        return (existing,);`,
-        `    }`,
-        `}`,
-      ].join('\n'),
-      functionsCalled: [
-        this.requireImport(...U128_FROM_FELT),
-        this.requireImport(...UINT256_LT),
-        arrayDef,
-        arrayLength,
-      ],
+      code: endent`
+        fn ${funcName}(warp_storage_array_ptr: felt252, index: u256) -> felt252 {
+            let length = ${lengthName}::read(warp_storage_array_ptr);
+            assert(index < length, 'Index out of bounds');
+            let existing = ${arrayName}::read(warp_storage_array_ptr, index);
+            if (existing == 0) {
+                let used = WARP_USED_STORAGE::read();
+                WARP_USED_STORAGE::write(used + ${valueCairoType.width});
+                ${arrayName}::write((warp_storage_array_ptr, index), used);
+                return used;
+            } else {
+                return existing;
```

Comment:
Since Cairo is very rust-like, and Cairo 1 devs are following the same guidelines that Rust follows, I think that our generated code should match it. You should change the `return expr;` for just `expr`